platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/network/NetworkPeerIdentifierTest.java

```diff
@@ -0,0 +1,134 @@
+
+        // sample node names from mainnet
+        final List<String> names =
+                List.of("node1", "node2", "node3", "node4", "node5", "node6", "node7", "node8", "node9", "node10");
+        // sample pfx file grabbed from mainnet
+        final KeyStore publicKeys = loadKeys(
+                ResourceLoader.getFile("preGeneratedKeysAndCerts/").resolve("publicMainnet.pfx"),
+                "password".toCharArray());
+        final PublicStores publicStores = PublicStores.fromAllPublic(publicKeys, names);
+
+        final List<PeerInfo> peerInfoList = new ArrayList<>();
+        for (int i = 0; i < 10; i++) {
+            final String name = names.get(i);
+            final NodeId node = new NodeId(i);
+            final PeerInfo peer = new PeerInfo(
+                    node,
+                    names.get(i),
+                    "127.0.0.1",
+                    Objects.requireNonNull(publicStores.getCertificate(KeyCertPurpose.SIGNING, name)));
+            peerInfoList.add(peer);
+        }
+
+        final PKIXParameters params = new PKIXParameters(publicStores.agrTrustStore());
+        final Set<TrustAnchor> trustAnchors = params.getTrustAnchors();
+
+        final Certificate[] certificates =
+                trustAnchors.stream().map(TrustAnchor::getTrustedCert).toArray(Certificate[]::new);
+        final PeerInfo matchedPeer = new NetworkPeerIdentifier().identifyTlsPeer(certificates, peerInfoList);
+        Assertions.assertNotNull(matchedPeer);
+    }
+
+    @Test
+    @DisplayName(
+            "asserts that when none of the peer's certificate matches any of the certs in the trust store, identifyTlsPeer returns null")
+    void testIdentifyTlsPeerReturnsNull()
+            throws URISyntaxException, KeyLoadingException, KeyStoreException, NoSuchAlgorithmException,
+                    NoSuchProviderException, KeyGeneratingException {
+        final List<String> names =
+                List.of("node1", "node2", "node3", "node4", "node5", "node6", "node7", "node8", "node9", "node10");
+        // sample pfx file grabbed from mainnet
+        final KeyStore publicKeys = loadKeys(
+                ResourceLoader.getFile("preGeneratedKeysAndCerts/").resolve("publicMainnet.pfx"),
+                "password".toCharArray());
+        final PublicStores publicStores = PublicStores.fromAllPublic(publicKeys, names);
+
+        // create list of peers with valid certs
+        final List<PeerInfo> peerInfoList = new ArrayList<>();
+        for (int i = 0; i < 10; i++) {
+            final String name = names.get(i);
+            final NodeId node = new NodeId(i);
+            final PeerInfo peer = new PeerInfo(
+                    node,
+                    names.get(i),
+                    "127.0.0.1",
+                    Objects.requireNonNull(publicStores.getCertificate(KeyCertPurpose.SIGNING, name)));
+            peerInfoList.add(peer);
+        }
```

Comment:
This is duplicated from the previous test. I think it would be cleaner to have a setup method that performs these tasks and makes the required variables available to all tests in the class. Then it is very easy for reader to see what is unique to each test, with the added bonus of reduced line count.

Reply:
agreed. Fixed